CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
 # Copyright (c) Microsoft Corporation.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+if (NOT DEFINED CMAKE_TOOLCHAIN_FILE AND EXISTS "${CMAKE_CURRENT_LIST_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake")
+    set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_LIST_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake")
+endif()

```

Comment:
It's weird that this is before `cmake_minimum_required`/`project`, which makes me suspect that it must be so. If that's the case, should we have a comment to that effect to avoid future maintainers "helpfully" moving this after `cmake_minimum_required`/`project`?

Reply:
Yes this is required before project is called. I don't think a comment is a bad idea.